addons/resources/src/ResourcePanel.js

```diff
@@ -0,0 +1,120 @@
+import { document } from 'global';
+import React, { Component } from 'react';
+import { Input, ActionButton } from '@storybook/components';
+import styled from '@emotion/styled';
+
+const storybookIframe = 'storybook-preview-iframe';
+
+const PanelWrapper = styled.div({
+  position: 'absolute',
+  top: '10px',
+  left: '10px',
+  width: '50%',
+});
+
+const PanelWrapperRow = styled.div({
+  display: 'flex',
+  'flex-direction': 'row',
+  padding: '3px',
+});
+
+export default class ResourcePanel extends Component {
+  constructor(props) {
+    super(props);
+    this.nodes = [];
+    this.state = { resources: [], addname: '' };
+  }
+
+  componentDidMount() {
+    this.iframe = document.getElementById(storybookIframe);
+    if (!this.iframe) {
+      throw new Error('Cannot find Storybook iframe');
+    }
+  }
+
+  componentDidUpdate() {
+    this.loadResources();
+  }
+
+  static getDerivedStateFromProps(props) {
+    return { resources: props.resources };
```

Comment:
it's called on every `setState` in React 16. So `state.resources` will always be the same as `props.resources`